README.md

```diff
@@ -64,13 +64,11 @@ One can still ask whether listening at lower volumes can at least be improved th
 
 There are a few more aspects of the plugin's configuration, that can be changed through the `asoundrc` file, although the default values should be fine in most cases.
 
-Compensation is done through FIR filtering, implemented via FFT convolution, where the filter coefficients, are recalculated each time one of the parameters (i.e. `Attenuation` or `Reference`) is changed.  The default filter length is 4096 taps, implemented via FFTs of length 16384, which should work well in most cases, but these values can be changed though the `window_length` and `fft_length` configuration variables respectively, if necessary.  One possible reason to do so, might be to lower the load on the CPU.
-
-Lowering the `window_length` would achieve that, but it might also cause insufficient boosting of the bass frequencies (which is also the case for the default length of 4096, as is visible in the plots above, albeit at frequencies well below the threshold of hearing).
+Compensation is done through FIR filtering, implemented via FFT convolution, where the filter coefficients, are recalculated each time one of the parameters (i.e. `Attenuation` or `Reference`) is changed.  The default filter length is 4096 taps, which should work well in most cases, but this value can be changed though the `impulse_length` configuration variable, if necessary.  One possible reason to do so, might be to lower the load on the CPU, but it might also cause insufficient boosting of the bass frequencies (which is also the case for the default length of 4096, as is visible in the plots above, albeit at frequencies well below the threshold of hearing).
 
 To examine the actual filter response calculated by the plugin, it should be recompiled with `make PLOT=yes && sudo make install`.  When built in this way, attempting to play through the plugin, will cause it to dump the filter parameters into a set of files in the current directory (named `parameters`, `sampled`, `realized`), along with a `gnuplot` script (named `filter.gnu`), which can be used to plot the generated filter response through the command `gnuplot -p filter.gnu`.
 
-Changing the `fft_length` parameter shouldn't have an effect on the quality of the filter, but it does affect computational efficiency.  Unfortunately, the optimal value depends on the ALSA period and buffer sizes, which are generally neither constant, nor in our control, so any tuning, must be made through trial and error.  Good values, will probably fall in the range of four to eight times the window length and be powers of two, or, at least, products of small prime factors.
+By default, the plugin determines the optimal value for the `fft_length` parameter, based on the maximum period size requested by the application. This behaviour can be overridden by setting the `fft_length` parameter to any non-zero value. For best performance, powers of two should be used. For mathematical correctness, the length should be greater than `impulse_length + period_length`.
```

Comment:
>But I don't think this is the case. If the FFT length is smaller, filtering can still be performed on a subset of the samples available. This is done here.

Fair enough, I didn't delve that deep into the code. That's inefficient though, requiring multiple FFTs to fill a single period without any benefit regarding latency.

Reply:
Well you do multiple FFTs anyway and in fact, multiple FFTs are done to speed things up (compared to a straight calculation  of the convolution with the impulse response).  If you're interested in efficiency considerations, see [here](https://en.wikipedia.org/wiki/Overlap%E2%80%93save_method#Efficiency_considerations).  If you had control over the period length, you could select it so as to minimize the calculations per sample (well approximately, since to reach the actual minimum you might need impractically large periods), but in any case we don't.  So the next best thing we can do, is to choose the FFT so as not to waste time and energy calculating more samples than the period.  This is the functionality you introduced.